src/app/clusters/temperature-control-server/temperature-control-server.cpp

```diff
@@ -0,0 +1,175 @@
+    }
+
+    CHIP_ERROR status = CHIP_NO_ERROR;
+
+    switch (aPath.mAttributeId)
+    {
+    case TemperatureSetpoint::Id: {
+        int16_t tempSetpoint = 0;
+        TemperatureSetpoint::Get(endpoint, &tempSetpoint);
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(tempSetpoint);
+        }
+        break;
+    }
+
+    case MinTemperature::Id: {
+        int16_t minTemperature = 0;
+        MinTemperature::Get(endpoint, &minTemperature);
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(minTemperature);
+        }
+        break;
+    }
+
+    case MaxTemperature::Id: {
+        int16_t maxTemperature = 0;
+        MaxTemperature::Get(endpoint, &maxTemperature);
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(maxTemperature);
+        }
+        break;
+    }
+
+    case Step::Id: {
+        int16_t step = 0;
+        Step::Get(endpoint, &step);
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(step);
+        }
+        break;
+    }
+
+    case CurrentTemperatureLevelIndex::Id: {
+        uint8_t currentTemperatureLevelIndex = 0;
+        CurrentTemperatureLevelIndex::Get(endpoint, &currentTemperatureLevelIndex);
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(currentTemperatureLevelIndex);
+        }
+        break;
+    }
+
+    default:
+        // We did not find a processing path, the caller will delegate elsewhere.
+        break;
+    }
+
+    return status;
+}
+
+bool TemperatureControlHasFeature(EndpointId endpoint, TemperatureControlFeature feature)
+{
+    bool success;
+    uint32_t featureMap;
+    success = (Attributes::FeatureMap::Get(endpoint, &featureMap) == EMBER_ZCL_STATUS_SUCCESS);
+
+    return success ? ((featureMap & to_underlying(feature)) != 0) : false;
+}
+
+static Status setTemperatureHandler(app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
+                                    int16_t targetTemperature, uint8_t targetTemperatureLevel)
+{
+    EndpointId endpoint = commandPath.mEndpointId;
+
+    if (TemperatureControlHasFeature(endpoint, TemperatureControlFeature::kTemperatureNumber))
+    {
+        int16_t minTemperature = 0;
+        int16_t maxTemperature = 0;
+        MinTemperature::Get(endpoint, &minTemperature);
+        MaxTemperature::Get(endpoint, &maxTemperature);
+        if (targetTemperature < minTemperature || targetTemperature > maxTemperature)
+        {
+            return Status::ConstraintError;
+        }
+        TemperatureSetpoint::Set(endpoint, targetTemperature);
+    }
+    if (TemperatureControlHasFeature(endpoint, TemperatureControlFeature::kTemperatureLevel))
+    {
+        // TODO: Check is targetTemperatureLevel is one of the SupportedTemperatureLevels
+        CurrentTemperatureLevelIndex::Set(endpoint, targetTemperatureLevel);
```

Comment:
I suspect this is wrong and the index should get set to the index of the relevant level, I expect.  But see https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/7005